### PR TITLE
Changes to start-release command to generate release-version.yml file

### PR DESCRIPTION
**Pull request by yharish991**
- Generates `release-version.yml` instead of `release.yml` from `release_template.yaml`
- Adds `version` to the release-version.yml by getting the version from the snowdrop bom.
- Adds `schedule` to release-version.yml, `start-release` command now accepts release and eol dates using `-r` and `-e` options. Calculates the due date by subtracting `1 month` from the release date.
- Some clean-up of the indentation in the files I touched.

Example command to test the `start-release` command
```
java -jar target/issues-manager-$(xpath -q -e  "/project/version/text()" pom.xml)-runner.jar \
    -u $JIRA_USERNAME \
    -p $JIRA_PASSOWRD \
    start-release \
    -g yharish991/spring-boot-bom/sb-2.4.x \
    -o $GIT_TOKEN \
    -r '2021-01-13' \
    -e '2021-11-13'
```

This PR is related to the issue https://app.zenhub.com/workspaces/snowdrop-5e13555db24025e26165460f/issues/snowdrop/team/364 , and the associate PR to make changes to `release_template.yml` file https://github.com/snowdrop/spring-boot-bom/pull/156

### CODE REVIEW

**Comment by metacosm** [comment]
Please avoid mixing formatting changes with other changes. Changing the format is fine as long as it happens in separate commits so that it doesn't interfere with the review process.

File: src/main/java/dev/snowdrop/release/model/Release.java [hunk 1]
```diff
@@ -114,16 +148,14 @@ public void setTest(boolean test) {
...
             }
         }
-        
-        // validate schedule
-        if (schedule == null) {
-            errors.add("missing schedule");
-        } else {
+
+        // validate schedule only if it not null
+        if (schedule != null) {
```
Comment (metacosm) [inline src/main/java/dev/snowdrop/release/model/Release.java:158]: I'd rather keep the previous check because now, if schedule is `null` for some reason, the code will proceed without error when it really should fail.

File: src/main/java/dev/snowdrop/release/services/GitService.java [hunk 2]
```diff
@@ -46,64 +46,61 @@
...
     }
-    
+
     public void commitAndPush(String commitMessage, File... changed) throws IOException {
         try {
             git.thenAcceptAsync(g -> {
                 try {
                     final var status = g.status().call();
                     final var uncommittedChanges = status.getUncommittedChanges();
+                    final var unTracked = status.getUntracked();
```
Comment (metacosm) [inline src/main/java/dev/snowdrop/release/services/GitService.java:96]:
```suggestion
                    final var untracked = status.getUntracked();
```

File: src/test/java/dev/snowdrop/release/services/ReleaseFactoryTest.java [hunk 3]
```diff
@@ -37,53 +37,24 @@
...
-        try {
-            factory.createFrom(getResourceAsStream("mismatched-version.yml"), getResourceAsStream("pom.xml"), true);
-            fail("should have failed on mismatched version");
-        } catch (IllegalArgumentException e) {
-            // expected
-            assertTrue(e.getMessage().contains("release version doesn't match"));
-        } catch (Throwable e) {
-            fail(e);
-        }
+    public void missingVersionShouldNotFail() throws Throwable {
```
Comment (metacosm) [inline src/test/java/dev/snowdrop/release/services/ReleaseFactoryTest.java:49]: Why did you change these tests? I understand that these cases shouldn't happen when using the tool as expected, however we cannot rule out that someone tries to process a manually edited release definition and I think that we should keep all the validation in place to account for that situation because we don't want to create product tickets using an invalid release definition.

**Review by metacosm — CHANGES_REQUESTED** [review]

File: src/test/java/dev/snowdrop/release/services/ReleaseFactoryTest.java (hunk 3 above)
Comment (yharish991) [reply src/test/java/dev/snowdrop/release/services/ReleaseFactoryTest.java:49]: @metacosm i made some changes to the code to always set `version` from the `pom.xml`, so there won't be a case where `version` is mismatched or is missing. https://github.com/snowdrop/issues-manager/pull/97/files#diff-c4475b0eb6572d8755db604107dfd103608fdf89816183921cf113950673c721R113

File: src/main/java/dev/snowdrop/release/model/Release.java (hunk 1 above)
Comment (yharish991) [reply src/main/java/dev/snowdrop/release/model/Release.java]: my latest commit addresses this

File: src/main/java/dev/snowdrop/release/services/GitService.java (hunk 2 above)
Comment (yharish991) [reply src/main/java/dev/snowdrop/release/services/GitService.java]: my latest commit addresses this

File: src/test/java/dev/snowdrop/release/services/ReleaseFactoryTest.java [hunk 4]
```diff
@@ -97,7 +68,7 @@ public void wrongScheduleShouldFail() {
             fail(e);
         }
     }
-    
+
     @Test
     public void mismatchedPOMVersionsShouldFail() {
         try {
```
[the remark this answers is not in the document]
Comment (yharish991) [reply src/test/java/dev/snowdrop/release/services/ReleaseFactoryTest.java:85]: my latest commit addresses this

**Comment by yharish991** [comment]
> Please avoid mixing formatting changes with other changes. Changing the format is fine as long as it happens in separate commits so that it doesn't interfere with the review process.

sorry about mixing the formatting changes, i understand it's a pain for reviewers. I was so excited about making formatting changes and pushed them along with actual changes.

**Review by jacobdotcosta — APPROVED** [review]
LGTM

**Comment by yharish991** [comment]
@metacosm can you re-review this when you get a chance ? :)

File: src/main/java/dev/snowdrop/release/model/Release.java [hunk 5]
```diff
@@ -99,10 +112,31 @@ public void setTest(boolean test) {
...
     }
-    
-    public List<String> validate(boolean skipProductRequests) {
+
+    public void setSchedule(String releaseDate, String eolDate) throws ParseException {
+        this.schedule = generateSchedule(releaseDate, eolDate);
+    }
+
+    public Schedule generateSchedule(String releaseDate, String eolDate) throws ParseException {
+        Schedule schedule = new Schedule();
```
Comment (metacosm) [inline src/main/java/dev/snowdrop/release/model/Release.java:121]: I'd rather we set the dates in the constructor than add setters as I don't think we'd need (or even allow) a schedule to be modified after it's been created. Immutability FTW!

File: src/main/java/dev/snowdrop/release/App.java [hunk 6]
```diff
@@ -78,68 +90,105 @@ public int run(String... args) throws Exception {
...
+            @CommandLine.Option(
+                    names = { "-t", "--test" },
+                    description = "Create a test release ticket using the SB project for all requests") boolean test,
+            @CommandLine.Option(
+                    names = { "-o", "--token" },
+                    description = "Github API token",
+                    required = true) String token,
+            @CommandLine.Option(
+                    names = { "-r", "--release-date" },
+                    description = "Release Date",
```
Comment (metacosm) [inline src/main/java/dev/snowdrop/release/App.java:163]: You should add the expected date format

File: src/main/java/dev/snowdrop/release/App.java [hunk 7]
```diff
@@ -78,68 +90,105 @@ public int run(String... args) throws Exception {
...
+                    required = true) String releaseDate,
+            @CommandLine.Option(
+                    names = { "-e", "--eol-date" },
+                    description = "EOL Date",
```
Comment (metacosm) [inline src/main/java/dev/snowdrop/release/App.java:167]: Add the date format and explicit what the EOL date means maybe?

**Review by metacosm — CHANGES_REQUESTED** [review]

**Comment by yharish991** [comment]
@metacosm pushed the changes to address your comments

**Review by metacosm — APPROVED** [review]
